code/go/0chain.net/blobbercore/challenge/protocol.go

```diff
@@ -70,49 +73,57 @@ func (cr *ChallengeEntity) LoadValidationTickets(ctx context.Context) error {
 		cr.StatusMessage = "No validators assigned to the challenge"
 		cr.UpdatedAt = time.Now().UTC()
 
+		zlogger.Logger.Debug(cr.StatusMessage + "for challenge: " + cr.ChallengeID)
+
 		if err := cr.Save(ctx); err != nil {
-			Logger.Error("[challenge]db: ", zap.String("challenge_id", cr.ChallengeID), zap.Error(err))
+			zlogger.Logger.Error("[challenge]db: ", zap.String("challenge_id", cr.ChallengeID), zap.Error(err))
+			return err
 		}
-		return common.NewError("no_validators", "No validators assigned to the challenge")
+		return nil
 	}
 
 	allocationObj, err := allocation.GetAllocationByID(ctx, cr.AllocationID)
 	if err != nil {
 		return err
 	}
 
+	// Lock allocation changes from happening in handler.CommitWrite function
+	allocMu := lock.GetMutex(allocationObj.TableName(), allocationObj.ID)
+	allocMu.Lock()
```

Comment:
@lpoli please add a comment on top of this function, that we should be unlocking it as soon as possible and never defer allocMu.Unlock() as it'll be concurrently required and if we defer, it'll be locked for longer time